Modules/gcmodule.c

```diff
@@ -1000,6 +1017,55 @@ show_stats_each_generations(struct _gc_runtime_state *state)
         buf, gc_list_size(&state->permanent_generation.head));
 }
 
+/* Deduce wich objects among "base" are unreachable from outside the list
+   and move them to 'unreachable'. The process consist in the following steps:
+
+1. Copy all references to a different field (gc_prev is used to hold this copy 
+   to save memory).
+2. Traverse all objects in "base" and visit all referred objects using
+   "tp_traverse" and for every visited object, substract 1 to the reference
+   count (the one that we copied in the previous step). After this step, all
+   objects that can be reached from outside must have positive reference count,
```

Comment:
change "from outside" by adding "directly" before it.
change "positive" to "strictly positive".